src/iris/api.py

```diff
@@ -1272,6 +1278,10 @@ def on_post(self, req, resp):
                     dynamic = step.get('dynamic_index') is not None
                     step['plan_id'] = plan_id
                     step['step'] = index
+                    # for backwards copatibility check if optional is not defined and set it to 0 if it isn't
+                    if 'optional' not in step:
```

Comment:
You can combine both map operations (the check and the assignment) to one using

    step.setdefault('optional', 0)

Reply:
Changed.